library/std/src/net/pipe/mod.rs

```diff
@@ -0,0 +1,280 @@
+                self.0.read_vectored(bufs)
+            }
+            fn is_read_vectored(&self) -> bool {
+                self.0.is_read_vectored()
+            }
+            fn read_to_end(&mut self, buf: &mut Vec<u8>) -> io::Result<usize> {
+                self.0.read_to_end(buf)
+            }
+            fn read_buf(&mut self, buf: io::BorrowedCursor<'_>) -> io::Result<()> {
+                self.0.read_buf(buf)
+            }
+        }
+    };
+}
+forward_io_read_traits!(PipeReader);
+forward_io_read_traits!(&PipeReader);
+
+macro_rules! forward_io_write_traits {
+    ($name:ty) => {
+        impl io::Write for $name {
+            fn write(&mut self, buf: &[u8]) -> io::Result<usize> {
+                self.0.write(buf)
+            }
+            fn flush(&mut self) -> io::Result<()> {
+                Ok(())
+            }
+
+            fn write_vectored(&mut self, bufs: &[io::IoSlice<'_>]) -> io::Result<usize> {
+                self.0.write_vectored(bufs)
+            }
+            fn is_write_vectored(&self) -> bool {
+                self.0.is_write_vectored()
+            }
+        }
+    };
+}
+forward_io_write_traits!(PipeWriter);
+forward_io_write_traits!(&PipeWriter);
+
+#[cfg(unix)]
+mod unix {
+    use super::*;
+
+    use crate::{
+        fs::File,
+        os::{
+            fd::{AsFd, AsRawFd, BorrowedFd, FromRawFd, IntoRawFd, OwnedFd, RawFd},
+            unix::fs::FileTypeExt,
+        },
+        sys::{
+            fd::FileDesc,
+            pipe::{anon_pipe, AnonPipe},
+        },
+        sys_common::{FromInner, IntoInner},
+    };
+
+    #[inline]
+    pub(super) fn pipe() -> io::Result<(PipeReader, PipeWriter)> {
+        anon_pipe().map(|(rx, tx)| (PipeReader(rx), PipeWriter(tx)))
+    }
+
+    macro_rules! impl_traits {
+        ($name:ty) => {
+            impl AsFd for $name {
+                fn as_fd(&self) -> BorrowedFd<'_> {
+                    self.0.as_fd()
+                }
+            }
+            impl AsRawFd for $name {
+                fn as_raw_fd(&self) -> RawFd {
+                    self.0.as_raw_fd()
+                }
+            }
+            impl From<$name> for OwnedFd {
+                fn from(pipe: $name) -> Self {
+                    FileDesc::into_inner(AnonPipe::into_inner(pipe.0))
+                }
+            }
+            impl FromRawFd for $name {
+                unsafe fn from_raw_fd(raw_fd: RawFd) -> Self {
+                    Self(AnonPipe::from_raw_fd(raw_fd))
+                }
+            }
+            impl IntoRawFd for $name {
+                fn into_raw_fd(self) -> RawFd {
+                    self.0.into_raw_fd()
+                }
+            }
+            impl From<$name> for Stdio {
+                fn from(pipe: $name) -> Self {
+                    Self::from(OwnedFd::from(pipe))
+                }
+            }
+        };
+    }
+    impl_traits!(PipeReader);
```

Comment:
Not sure if it is the right places as it push hook warns me to not use `cfg(unix)` here.

I checked and it seems that `AsFd` implementation is put inside `owned_fd` mod.

However I don't think I can put `TryFrom<OwnedFd> for PipeReader` in there

Reply:
Figured it out, moving it to `src/sys` fixed it